ovs/extensions/hypervisor/apis/vmware/sdk.py

```diff
@@ -903,13 +903,82 @@ def get_nfs_datastore_object(self, ip, mountpoint, filename, host=None):
+
+        task = self._client.service.CreateVirtualDisk_Task(self._serviceContent.virtualDiskManager, disk_url, None, config)
+
+        if wait:
+            self.wait_for_task(task)
+        return disk_path.lstrip('/')
+
+    @authenticated()
+    def delete_disk(self, ip, mountpoint, disk_path, host=None, wait=True):
+        esxhost = self._validate_host(host)
+        datastore = self.get_datastore(ip, mountpoint, host=esxhost)
+        if not datastore:
+            raise RuntimeError('Could not find datastore')
+        if not self.file_exists(ip, mountpoint, disk_path):
+            logger.warning('Disk {0} already deleted'.format(disk_path))
+            return
+        disk_url = '[{0}] {1}'.format(datastore.name, disk_path)
+
+        task = self._client.service.DeleteVirtualDisk_Task(self._serviceContent.virtualDiskManager, disk_url, None)
+        if wait:
+            self.wait_for_task(task)
+
+    def _get_files_on_datastore(self, ip, mountpoint, host=None):
+        esxhost = self._validate_host(host)
+        datastore = self.get_datastore(ip, mountpoint, host=esxhost)
+        if not datastore:
+            raise RuntimeError('Could not find datastore')
+        hdbss = self._client.factory.create('ns0:HostDatastoreBrowserSearchSpec')
+        hdbss.matchPattern = "*.*"
+        task = self._client.service.SearchDatastore_Task(datastore.browser, '[{0}]'.format(datastore.name), hdbss)
+        result = self.get_task_info(task)
+        if result.info.state == 'success':
+            return [file.path for file in result.info.result.file]
+        logger.warning(result.info.error.localizedMessage)
+        return []
+
+    @authenticated()
+    def extend_disk(self, ip, mountpoint, disk_path, size, host=None, wait=True):
+        if host is None:
+            host = self._esxHost
+        esxhost = self._validate_host(host)
+        datastore = self.get_datastore(ip, mountpoint, host=esxhost)
+        if not datastore:
+            raise RuntimeError('Could not find datastore')
+        if not self.file_exists(ip, mountpoint, disk_path):
+            raise RuntimeError('Disk {0} must be created first'.format(disk_path))
+        disk_url = '[{0}] {1}'.format(datastore.name, disk_path)
+        capacityKb = size * 1024 * 1024
+
+        task = self._client.service.ExtendVirtualDisk_Task(self._serviceContent.virtualDiskManager, disk_url, None, capacityKb)
+
+        if wait:
+            self.wait_for_task(task)
+
     def file_exists(self, ip, mountpoint, filename):
         try:
             self.get_nfs_datastore_object(ip, mountpoint, filename)
             return True
-        except Exception, ex:
-            logger.debug('File does not exist: {0}'.format(ex))
-            return False
+        except Exception as ex:
+            logger.debug('Exception in get_nfs_datastore_object: {0}'.format(ex))
+            files_on_datastore = self._get_files_on_datastore(ip, mountpoint)
```

Comment:
it's easier to test if separate, also it could be used in other methods as well.